fabtests/pytest/common.py

```diff
@@ -455,19 +455,26 @@ def prepare_base_command(self, command_type, executable,
         if "PYTEST_XDIST_WORKER" in os.environ:
             worker_id = int(os.environ["PYTEST_XDIST_WORKER"].replace("gw", ""))
             hmem_device_id = worker_id % num_hmem
-            if host_memory_type == "cuda":
-                command += " -i {}".format(hmem_device_id)
-            else:
-                assert host_memory_type == "neuron"
-                num_cores = num_neuron_cores_on_device(host_ip, hmem_device_id)
+        else:
+            hmem_device_id = 0
+
+        if host_memory_type == "cuda":
+            command += " -i {}".format(hmem_device_id)
+        else:
+            assert host_memory_type == "neuron"
+            num_cores = num_neuron_cores_on_device(host_ip, hmem_device_id)
+            if command_type == "server":
                 additional_environment = "NEURON_RT_VISIBLE_CORES={}".format(
```

Comment:
I remember you can specify multiple cores for single process. Does it you work if you make both client and server use something like `NEURON_RT_VISIBLE_CORES=hmem_device_id * num_cores ,hmem_device_id * num_cores + 1`?

Reply:
No. The server takes both cores.